src/System.Linq.Expressions/src/System/Dynamic/ExpandoObject.cs

```diff
@@ -1113,3 +1114,86 @@ private static int GetAlignedSize(int len)
     }
 }
 
+namespace System.Runtime.CompilerServices
+{
+    //
+    // Note: these helpers are kept as simple wrappers so they have a better 
+    // chance of being inlined.
+    //
```

Comment:
I read it as meaning that if inlinabilty analysis ever included the size of the caller, they'd then benefit,

If inlining was desired it could perhaps be done at the expression creation: See if the more direct `typeof(ExpandoObject).GetMethod("TryGetValue", BindingFlags.NonPublic | BindingFlags.Instance)` works, and use it if it does.

That'd be spending a guinea to save a farthing, but if combined with @bartdesmet's idea of caching methodinfos that get used a lot, the cost-benefit changes a lot.